go-controller/pkg/ovn/pods.go

```diff
@@ -122,24 +123,28 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod, portInfo *lpInfo) (err er
 			}
 			return fmt.Errorf("unable to unmarshal pod annotations for pod %s/%s: %w", pod.Namespace, pod.Name, err)
 		}
+		nodeName = pod.Spec.NodeName
```

Comment:
we should iterate through the switches and find the right switch in nbdb cache in this case

Reply:
ack; will do.